pkgs/by-name/ma/matrix-continuwuity/package.nix

```
@@ -13,35 +13,94 @@
   enableBlurhashing ? true,
   # upstream continuwuity enables jemalloc by default, so we follow suit
   enableJemalloc ? true,
-  rust-jemalloc-sys,
+  rust-jemalloc-sys-unprefixed,
   enableLiburing ? stdenv.hostPlatform.isLinux,
   liburing,
   nixosTests,
 }:
 let
-  rust-jemalloc-sys' = rust-jemalloc-sys.override {
-    unprefixed = !stdenv.hostPlatform.isDarwin;
-  };
-  rocksdb' = rocksdb.override {
-    inherit enableLiburing;
-    # rocksdb does not support prefixed jemalloc, which is required on darwin
-    enableJemalloc = enableJemalloc && !stdenv.hostPlatform.isDarwin;
-    jemalloc = rust-jemalloc-sys';
-  };
+  rocksdb' =
+    (rocksdb.override {
+      inherit enableLiburing;
+      # rocksdb does not support prefixed jemalloc, which is required on darwin
+      enableJemalloc = enableJemalloc && !stdenv.hostPlatform.isDarwin;
+      jemalloc = rust-jemalloc-sys-unprefixed;
+    }).overrideAttrs
+      (
+        final: old: {
+          version = "10.4.fb";
+          src = fetchFromGitea {
+            domain = "forgejo.ellis.link";
+            owner = "continuwuation";
+            repo = "rocksdb";
+            rev = "10.4.fb";
+            hash = "sha256-/Hvy1yTH/0D5aa7bc+/uqFugCQq4InTdwlRw88vA5IY=";
+          };
+
+          patches = [ ];
+
+          cmakeFlags =
+            lib.subtractLists [
+              # no real reason to have snappy or zlib, no one uses this
+              "-DWITH_SNAPPY=1"
+              "-DZLIB=1"
+              "-DWITH_ZLIB=1"
+              # we dont need to use ldb or sst_dump (core_tools)
+              "-DWITH_CORE_TOOLS=1"
+              # we dont need to build rocksdb tests
+              "-DWITH_TESTS=1"
+              # we use rust-rocksdb via C interface and dont need C++ RTTI
+              "-DUSE_RTTI=1"
+              # this doesn't exist in RocksDB, and USE_SSE is deprecated for
+              # PORTABLE=$(march)
+              "-DFORCE_SSE42=1"
+            ] old.cmakeFlags
+            ++ [
+              # no real reason to have snappy, no one uses this
+              "-DWITH_SNAPPY=0"
+              "-DZLIB=0"
+              "-DWITH_ZLIB=0"
+              # we dont need to use ldb or sst_dump (core_tools)
+              "-DWITH_CORE_TOOLS=0"
+              # we dont need trace tools
+              "-DWITH_TRACE_TOOLS=0"
+              # we dont need to build rocksdb tests
+              "-DWITH_TESTS=0"
+              # we use rust-rocksdb via C interface and dont need C++ RTTI
+              "-DUSE_RTTI=0"
+              "-DPORTABLE=1"
+            ];
+          outputs = [ "out" ];
+          preInstall = "";
+          postPatch = ''
+            # Fix gcc-13 build failures due to missing <cstdint> and
+            # <system_error> includes, fixed upstyream sice 8.x
```

```suggestion
          postPatch =
          # Fix gcc-13 build failures due to missing <cstdint> and
          # <system_error> includes, fixed upstyream sice 8.x
          ''
```
Nit: prefer having comments outside of nix strings. It allows changing them without rebuilding anything.

---

I had copied this from upstream, turns out we do not need the modified postPatch so I removed it.